### ISSUE TITLE
Some units troubles in the dialog box of a component

### ISSUE THREAD

**Issue by mpradovelasco** [OpenModelica/OpenModelica#10819]
### Description

This issue tries to describe the potential bugs commented in https://github.com/OpenModelica/OpenModelica/discussions/10776#discussioncomment-6135743 .

I use OMedit 1.22-dev v1.22.0-dev-161-g8eff86a43d on Windows 11. "Enable Instance API" is OFF (since if I set ON then I do not see units in the dialog box). 

I use the single model defined as:
```
package TestdisplayUnits
  type MTest = Real (final quantity = "Test",
    final unit = "kg",
    displayUnit = "ug");
  model M 
    parameter MTest x = 1.5;
    parameter MTest xa[2] = {2.5,3.0};
    parameter MTest xad[2,2] = {{1.0,2.2},{3.3,4.3}};
    annotation(
      Diagram(graphics = {Rectangle(fillColor = {255, 255, 255}, fillPattern = FillPattern.Solid, 
                          extent = {{-100, 100}, {100, -100}})}));
  end M;
  model S
    M m annotation(
      Placement(visible = true, transformation(origin = {4, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  end S;
```
i) When double click in component m of the diagram s, the xad parameter appear with SI units (kg), but displaying ug units, what causes confusion. This is a double Real array. 

![image](https://github.com/OpenModelica/OpenModelica/assets/114144098/f9edb095-c344-4141-8b0b-4bb7356aa9f4)

ii) When I start to write a value with the format { , }, the displaying units are frozen in the SI (kg in this case), as shown for xa parameter:

![image](https://github.com/OpenModelica/OpenModelica/assets/114144098/1267bc96-4dd3-43fc-9901-b165e36d88a9)

The same occurs if I begin to write any a letter or any different to a number by mistake. I need to close and reopen the dialog to start again. 
It seems not possible to introduce arrays parameters in non SI units, since the display unit modifies itself to SI.

iii) Once the array parameters are introduced in SI, as xa and xad in the following figure, the xa (single array) display units may be modified to a non SI unit (ug in figure), but the double array (xad) cannot be modified (it is frozen in "kg"). 

![image](https://github.com/OpenModelica/OpenModelica/assets/114144098/cb644a5f-87d6-4e5d-aa26-59eec988b5e0)

Thanks for the support

**Comment by adeas31** [comment]
I have no idea why you can't see the units.

Is it possible to have some zoom or teams session where we can discuss this issue?

**Comment by casella** [comment]
@mpradovelasco I think I know what the problem is. Display units are meant to allow the input of parameter values with convenient engineering units, which are then automatically converted to SI units, that are written in the modifiers of the Modelica code. This is supposed to work when you type in `literal values`, i.e., plain numbers.

We agreed some time ago that if you type in _expressions_, which may involve other parameters (that have SI units), applying unit conversion becomes ambiguous and dangerous, so we should disable it and stick to the (SI) unit.

The problem in this case is that you need to input an _array_ of values, so there are those curly braces, which OMEdit probably interprets as an expression.

This issue should have been fixed long time ago in #4840, but maybe there is still something wrong there, maybe due to the new instance-based API. @adeas31 can you please check?

**Comment by mpradovelasco** [comment]
Thank you @casella and @adeas31. I will test again with the latest dev when you had could checked that issue.
